.github/workflows/build-deploy-pudl.yml

```diff
@@ -25,27 +25,39 @@ jobs:
       - name: Use pudl-deployment-dev vm and dev branch if running on a schedule
         if: ${{ (github.event_name == 'schedule') }}
         run: |
-          echo "This action was triggered by a schedule." && echo "GCE_INSTANCE=pudl-deployment-dev" >> $GITHUB_ENV && echo "GITHUB_REF=dev" >> $GITHUB_ENV
+          echo "This action was triggered by a schedule."
+          echo "GCE_INSTANCE=pudl-deployment-dev" >> $GITHUB_ENV
+          echo "GCE_INSTANCE: $GCE_INSTANCE"
+          echo "GITHUB_REF=dev" >> $GITHUB_ENV
+          echo "GITHUB_REF: $GITHUB_REF"
 
       - name: Checkout Repository
         uses: actions/checkout@v4
         with:
           ref: ${{ env.GITHUB_REF }}
 
-      - name: Get HEAD of the branch (main or dev)
+      - name: Set action environment variables
         run: |
-          echo "ACTION_SHA=$(git rev-parse HEAD)" >> $GITHUB_ENV
+          echo "NIGHTLY_TAG=nightly-$(date +%Y-%m-%d)" >> $GITHUB_ENV
+          echo "NIGHTLY_TAG: $NIGHTLY_TAG"
           echo "SHORT_SHA=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+          echo "SHORT_SHA: $SHORT_SHA"
+          echo "BUILD_TIMESTAMP=$(date +%Y-%m-%d-%H%M)" >> $GITHUB_ENV
+          echo "BUILD_TIMESTAMP: $BUILD_TIMESTAMP"
+          echo "BUILD_ID=${BUILD_TIMESTAMP}-${SHORT_SHA}-${GITHUB_REF}
+          echo "BUILD_ID: $BUILD_ID"
```

Comment:
The `BUILD_ID` was being constructed in a couple of different ways in different places which was breaking things. See #3171.  Now it's constructed here and passed around as a unified entity.